dbt/adapters/bigquery/relation_configs/_options.py

```diff
@@ -137,6 +141,7 @@ def parse_bq_table(cls, table: BigQueryTable) -> Dict[str, Any]:
             "refresh_interval_minutes": table.mview_refresh_interval.seconds / 60,
             "expiration_timestamp": table.expires,
             "max_staleness": None,
+            "allow_non_incremental_definition": table._properties.get("materializedView", {}).get("allowNonIncrementalDefinition"),
```

Comment:
We try to not use private attributes/methods (`_properties`) where we can. Is there another way to parse this from the `BigQueryTable` object?

Reply:
Unfortunately I don't think so, I mentioned this in the PR body

> The only wrinkle is that the Python client library doesn't yet have a property for this variable, so it's being pulled directly from the properties dict instead.

I'll make an issue on the `google-cloud-bigquery` repo as well, would you consider this a blocker or is it good enough to follow up and replace once it's added?